modopt/opt/proximity.py

```diff
@@ -434,3 +435,104 @@ def _cost_method(self, *args, **kwargs):
 
         return np.sum([operator.cost(data) for operator, data in
                        zip(self.operators, args[0])])
+
+
+class OrderedWeightedL1Norm(ProximityParent):
+    r"""Ordered Weighted L1 norm proximity operator
+
+    This class defines the OWL proximity operator described in [F2014]
+
+    Parameters
+    ----------
+    weights : np.ndarray
+        Weights values they should be sorted in a non-increasing order
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from modopt.opt.proximity import OrderedWeightedL1Norm
+    >>> A = np.arange(5)*5
+    array([ 0,  5, 10, 15, 20])
+    >>> prox_op = OrderedWeightedL1Norm(np.arange(5))
+    >>> prox_op.weights
+    array([4, 3, 2, 1, 0])
+    >>> prox_op.op(A)
+    array([ 0.,  4.,  8., 12., 16.])
+    >>> prox_op.cost(A, verbose=True)
+     - OWL NORM (X): 150
+    150
+    """
+
+    def __init__(self, weights):
+
+        if any([weights_i < 0 for weights_i in np.squeeze(weights)]):
```

Comment:
you could do something like:
```python
(np.squeeze(weights) < 0).any()
```

Also since you are using `np.squeeze(weights)` a bit after, you might as well create a variable for it.

Reply:
Changing it to flatten